src/base/rss/rssfeed.cpp

```diff
@@ -399,6 +399,22 @@ void Feed::downloadArticleTorrentIfMatching(const ArticlePtr &article)
     BitTorrent::Session::instance()->addTorrent(torrentUrl, params);
 }
 
+QString Feed::generateSavePath(const QString &savePathTpl, const ArticlePtr &article)
+{
+    QRegExp rx("<(.+)>");
+    QString result = savePathTpl;
+    int pos = 0;
+    while ((pos = rx.indexIn(result, pos)) != -1) {
```

Comment:
Under certain conditions there may be an infinite loop.
You should increase pos by replacement length (`pos += article->getValue(rx.cap(1))`) at the end of each iteration.